adapter/utils/src/main/java/org/mobilitydata/gtfsvalidator/utils/GeodeticUtils.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.utils;
+
+import org.locationtech.spatial4j.distance.DistanceCalculator;
+import org.locationtech.spatial4j.shape.Point;
+import org.locationtech.spatial4j.shape.ShapeFactory;
+
+import static org.locationtech.spatial4j.context.SpatialContext.GEO;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceCalculationUtils;
+import org.mobilitydata.gtfsvalidator.usecase.distancecalculationutils.DistanceUnit;
+
+import java.util.Map;
+
+/**
+ * Interface implementation using external library org.locationtech.spatial4j. This is used to compute distance between
+ * two {@code ShapePoint}.
+ */
+public class GeodeticUtils implements DistanceCalculationUtils {
+
+    /**
+     * Return the distance between two points given there lat/lon positions. Note that points of origin and destination
+     * can be swapped.
+     *
+     * @param originLatitude        latitude of the origin point
+     * @param destinationLatitude   latitude of the destination point
+     * @param originLongitude       longitude of the origin point
+     * @param destinationLongitude  longitude of the destination point
+     * @param distanceUnit          unit of the desired result of computation
+     * @return the distance between two points given there lat/lon positions in the specified unit
+     */
+    @Override
+    public double distanceBetweenTwoPoints(final float originLatitude, final float destinationLatitude,
```

Comment:
Typically lat and lon for the same point are kept adjacent in method signatures - so I suggest changing this to:

```java
    public double distanceBetweenTwoPoints(final float originLatitude, final float originLongitude,
                                                  final float destinationLatitude,
                                                  final float destinationLongitude,
                                                  final DistanceUnit distanceUnit) {
```